### PR TITLE
Improve Batches Search Query

### PR DESCRIPTION
**Pull request by charlie-tf**
Use multiple query params instead of a comma-separated list in one param.

### CODE REVIEW

File: src/components/InventoryV2/InventoryListByBatch.tsx [hunk 1]
```diff
@@ -132,14 +127,15 @@ export default function InventoryListByBatch({ setReportData }: InventoryListByB
...
   const updateFilterQueryParams = useCallback(
     (f: InventoryFiltersType) => {
-      query.set('facilityId', f.facilityIds?.map((n) => n.toString())?.join(',') ?? '');
-      query.set(
-        'subLocationName',
-        f.subLocationsIds?.map((id) => subLocations?.find((sl) => sl.id === id)?.name ?? '')?.join(',') ?? ''
+      const newQuery = new URLSearchParams();
+      f.facilityIds?.forEach((n) => newQuery.append('facilityId', n.toString()));
+      f.subLocationsIds?.forEach((id) =>
+        newQuery.append('subLocationName', subLocations?.find((sl) => sl.id === id)?.name ?? '')
```
Comment (karthikbtf) [inline src/components/InventoryV2/InventoryListByBatch.tsx:133]: optionally sublocationIds->map to name->filter on non-empty name->foreach append to newQuery

**Review by karthikbtf — APPROVED** [review]
Awesome, thanks so much 👍

File: src/components/InventoryV2/InventoryListByBatch.tsx [hunk 2]
```diff
@@ -93,21 +94,15 @@ export default function InventoryListByBatch({ setReportData }: InventoryListByB
...
   const [subLocationNames, setSubLocationNames] = useState<string[]>([]);
   const [nurseryIds, setNurseryIds] = useState<number[]>([]);
   const query = useQuery();
   useEffect(() => {
-    const querySubLocationNames = query
-      .get('subLocationName')
-      ?.trim()
-      ?.split(',')
-      ?.filter((s) => s !== '');
+    const querySubLocationNames = query.getAll('subLocationName')?.filter((s) => s !== '');
```
[the remark this answers is not in the document]
Comment (charlie-tf) [reply src/components/InventoryV2/InventoryListByBatch.tsx:102]: Sure

[the remark this answers is not in the document]
Comment (charlie-tf) [reply src/components/InventoryV2/InventoryListByBatch.tsx:102]: Well... I'm not sure that the filter isn't necessary. It's needed to protect from a case where the url params are `?facilityId=&subLocationName=`, which will yield empty strings in the list.

[the remark this answers is not in the document]
Comment (karthikbtf) [reply src/components/InventoryV2/InventoryListByBatch.tsx:102]: Yes, makes sense. I thought this was mostly to protect from how we were setting the url params. The user can certainly edit the url.

File: src/components/InventoryV2/InventoryListByBatch.tsx (hunk 1 above)
Comment (charlie-tf) [reply src/components/InventoryV2/InventoryListByBatch.tsx:133]: Okay, yes, makes sense to filter empty names here

**Review by karthikbtf — APPROVED** [review]

**Pull request merged**